internal/uuid/uuid.go

```diff
@@ -20,3 +27,80 @@ func New(r io.Reader) (string, error) {
 	bs[6] = bs[6]&^0xf0 | 0x40
 	return fmt.Sprintf("%x-%x-%x-%x-%x", bs[0:4], bs[4:6], bs[6:8], bs[8:10], bs[10:]), nil
 }
+
+// Parse will use the google/uuid library to parse the string into a uuid
+// if parsing fails, it will return an empty map. It will fill the map
+// with some decoded values with fillMap
+func Parse(s string) (map[string]interface{}, error) {
+	uuid, err := uuid.Parse(s)
+	if err != nil {
+		return nil, err
+	}
+	out := make(map[string]interface{}, getVersionLen(int(uuid.Version())))
+	fillMap(out, uuid)
+	return out, nil
+}
+
+// Fills the map with values from the uuid. Version and variant for every version.
+// Version 1-2 has decodable values that could be of use, version 4 is random,
+// and version 3,5 is not feasible to extract data.
+func fillMap(m map[string]interface{}, u uuid.UUID) {
+	m["version"] = int(u.Version())
+	m["variant"] = u.Variant().String()
+	switch version := m["version"]; version {
+	case 1, 2:
+		m["time"] = nanoUnix(u.Time())
+		m["nodeid"] = macVars(u.NodeID()[0]) + ":" + byteDecimalToHex(u.NodeID())
+		m["clocksequence"] = u.ClockSequence()
+		if version == 2 {
+			m["id"] = int(u.ID())
+			m["domain"] = u.Domain().String()
+		}
+	}
+}
+
+// macVars will take the first byte of a MAC-address and check for the
+// local/global bit and check for the unicast/multicast bit of the byte,
+// and return a string with this info.
+func macVars(inpb byte) string {
+	switch {
+	case inpb&byte(0b00000011) == byte(0b00000011):
+		return "local:multicast"
+	case inpb&byte(0b00000001) == byte(0b00000001):
+		return "global:multicast"
+	case inpb&byte(0b00000010) == byte(0b00000010):
+		return "local:unicast"
+	case inpb&byte(0b00000000) == byte(0b00000000):
```

Comment:
Would this ever be false? I might be missing some bit fiddling edge case, but & with 0 is 0, isn't it?
```suggestion
	default:
```

We might also just put "global:unicast" into the return below and drop the default branch.

Related question: are these prefixes common? If so, we should reference something, maybe, like an RFC? 🤔

Reply:
It seems to me like the preferred format of a human readable MAC address should be groups of two hexdigits separated by dashes, for example`01-23-45-67-89-AB`, ref https://en.wikipedia.org/wiki/MAC_address#Notational_conventions, so I say that we should go for that format.

The prefixes here are not standard, but do appear in some online decoders, such as https://www.uuidtools.com/decode

It is hard to anticipate how this builtin will be used. Will the "I/G" (the prefix here) be used by many? I doubt it. So perhaps the specially interested can compute it themselves by extracting the specific hexadecimal in Rego and parsing it?

Another alternative is to expose this information as yet another field on the returned object. Low cost, but is it necessary? Maybe not.

Reply:
That is correct. AND-ing with 0 is 0, so just made it a return without the default. It wont be false.
As Kristian pointed out, the most common way to display MAC addresses is groups of hexdigits separated by dashes. So made a change to it with #203911 to do that, and extracted the `"unicast/multicast"` and `"local/global"` identifiers to another key in the object if someone would want to see them

Reply:
That sounds like a good compromise, thanks!